app/controllers/users_controller.rb

```diff
@@ -0,0 +1,43 @@
+class UsersController < ApplicationController
+  def index; end
+
+  def show; end
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      redirect_to @user
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  def edit
+    @user = User.find(params[:id])
+  end
+
+  def update
+    @user = User.find(params[:id])
+    if @user.update(user_params)
+      redirect_to @user
+    else
+      render :edit, status: :unprocessable_entity
+    end
+  end
+
+  def destroy
+    @user = User.find(params[:id])
+    @user.destroy
+    redirect_to users_path
+  end
+
+  private
+
+  def user_params
+    params.require(:user).permit(:name, :bio, :photo)
+  end
```

Comment:
- [ ] I looked for the request `spec/request` or `spec/controller` folder that should contain all the implementation of the below project requirements 👇🏾 but could not find it. Please add a test file for each controller. 

![image](https://github.com/KAYEMBAJONATHAN/Blog-app-/assets/84156714/6a080025-978f-4af6-b5b5-3f90c9f2d5e9)

Below here is example and not the actual solution, but your solution should look like this for all controllers. 

```ruby
require 'rails_helper'

RSpec.describe "Products", type: :request do
  describe "GET /products" do
    it "returns a successful response" do
      get "/products"
      expect(response).to have_http_status(:success)
    end

    it "renders the correct template" do
      get "/products"
      expect(response).to render_template(:index)
    end

    it "includes correct placeholder text in the response body" do
      get "/products"
      expect(response.body).to include("Placeholder text")
    end
  end
end
```